src/main/java/vazkii/botania/common/block/dispenser/BehaviourStick.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * This class is distributed as part of the Botania Mod.
+ * Get the Source Code in github:
+ * https://github.com/Vazkii/Botania
+ *
+ * Botania is Open Source and distributed under the
+ * Botania License: http://botaniamod.net/license.php
+ */
+package vazkii.botania.common.block.dispenser;
+
+import net.minecraft.block.DispenserBlock;
+import net.minecraft.dispenser.DefaultDispenseItemBehavior;
+import net.minecraft.dispenser.IBlockSource;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.Direction;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+
+import vazkii.botania.common.item.ItemObedienceStick;
+
+import javax.annotation.Nonnull;
+
+public class BehaviourStick extends DefaultDispenseItemBehavior {
+
+	@Nonnull
+	@Override
+	protected ItemStack dispenseStack(IBlockSource source, ItemStack stack) {
+		World world = source.getWorld();
+		Direction facing = world.getBlockState(source.getBlockPos()).get(DispenserBlock.FACING);
+		BlockPos pos = source.getBlockPos().offset(facing);
+
+		if (ItemObedienceStick.applyStick(world, pos)) {
+			return stack;
+		}
+
+		return super.dispenseStack(source, stack);
```

Comment:
So this launches the stick out of the dispenser if it fails? I'm pretty sure vanilla behaviour is to do nothing if the target is not eligible (like buckets)
See OptionalDispenseBehaviour or whatever it's called. The one with a setSuccess method